test/e2e/cni.go

```diff
@@ -84,13 +85,24 @@ func InstallCalicoHelmChart(ctx context.Context, input clusterctl.ApplyClusterTe
 	By("Waiting for Ready calico-node daemonset pods")
 	for _, ds := range []string{"calico-node"} {
 		waitInput := GetWaitForDaemonsetAvailableInput(ctx, clusterProxy, ds, CalicoSystemNamespace, specName)
+		waitInput.PodsRunning = int32(*input.ConfigCluster.WorkerMachineCount) + int32(*input.ConfigCluster.ControlPlaneMachineCount)
 		WaitForDaemonset(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-daemonset")...)
 	}
 	// TODO: enable this for all clusters once calico for windows is part of the helm chart.
 	if hasWindows {
 		By("Waiting for Ready calico windows pods")
-		for _, ds := range []string{"calico-node-windows", "kube-proxy-windows"} {
+		for _, ds := range []string{"calico-node-windows"} {
 			waitInput := GetWaitForDaemonsetAvailableInput(ctx, clusterProxy, ds, CalicoSystemNamespace, specName)
+			windowsNodeCount, _ := strconv.ParseInt(os.Getenv("WINDOWS_WORKER_MACHINE"), 10, 64)
+			waitInput.PodsRunning = int32(windowsNodeCount)
+			WaitForDaemonset(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-daemonset")...)
+		}
+
+		By("Waiting for Ready kube-proxy windows pods")
+		for _, ds := range []string{"kube-proxy-windows"} {
+			waitInput := GetWaitForDaemonsetAvailableInput(ctx, clusterProxy, ds, "kube-system", specName)
```

Comment:
there should be constant somewhere for kube-system you can use

Reply:
I dropped this change for now

Reply:
Moved it to https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2933